### PR TITLE
Krn ticket21 Post menu and details

### LINKED ISSUE #21: Post Details
As a post connoisseur I would like to view details of a post so that I can focus on a particular post and not the entire list of posts.

**Given** the user is viewing the Post Management menu  
**When** they select the option to view post details  
**Then** they should be presented with a list of posts to choose from  

**Given** the user chooses a post  
**When** they enter the selection and hit enter  
**Then** the Post Details menu should be displayed

**Given** the user wishes to view the post details  
**When** they select "View"  
**Then** the post's title, url and publication date should be displayed  

The Post Detail menu should have the following options:

- View
- Add Tag
- Remove Tag
- Note Management
- Return

> **NOTE:** The other menu options will be outlined in future stories.

### PR DESCRIPTION
**Pull request by karlarn**
# Description

Created post management file to give user and option to view details

Fixes # (issue)

Closes #21

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# Testing Instructions
 go through post management to select post details to select view on the menu all other options do not function  

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings or errors
- [X] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by ToryT23 — DISMISSED** [review]
looks good

**Review by jandershock — CHANGES_REQUESTED** [review]

File: TabloidCLI/Repositories/PostRepository.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public List<Post> GetAll()
                                 {
                                     Id = reader.GetInt32(reader.GetOrdinal("BlogId"))
                                 }
-                            });
+                            }); ;
```
[the remark this answers is not in the document]
Comment (joshdbarton) [reply TabloidCLI/Repositories/PostRepository.cs:40]: yep

File: TabloidCLI/Repositories/PostRepository.cs [hunk 2]
```diff
@@ -47,8 +47,49 @@ public List<Post> GetAll()
...
+                    cmd.CommandText = @"SELECT *
+                                          FROM Post 
+                                         WHERE Id = @id";
+
+                    cmd.Parameters.AddWithValue("@id", id);
+
+                    Post post = null;
+
+                    SqlDataReader reader = cmd.ExecuteReader();
+                    while (reader.Read())
```
Comment (joshdbarton) [inline TabloidCLI/Repositories/PostRepository.cs:64]: while loop for one row?

File: TabloidCLI/Repositories/PostRepository.cs [hunk 3]
```diff
@@ -47,8 +47,49 @@ public List<Post> GetAll()
...
+                            post = new Post()
+                            {
+                                Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                                Title = reader.GetString(reader.GetOrdinal("Title")),
+                                Url = reader.GetString(reader.GetOrdinal("Url")),
+                                PublishDateTime = reader.GetDateTime(reader.GetOrdinal("PublishDateTime")),
+                            };
+                        }
+                        //this may be helpful later
+                        //if (!reader.IsDBNull(reader.GetOrdinal("TagId")))
```
Comment (joshdbarton) [inline TabloidCLI/Repositories/PostRepository.cs:77]: let's take out the zombie code. If you need to get it from the Author repo later to copy over, it's still there.

File: TabloidCLI/Repositories/PostRepository.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (karlarn) [reply TabloidCLI/Repositories/PostRepository.cs]: Removed.

File: TabloidCLI/UserInterfaceManagers/PostDetailManager.cs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        {
+            Post post = _postRepository.Get(_postId);
+            Console.WriteLine($"Title: {post.Title}");
+            Console.WriteLine($"Url: {post.Url}");
+            Console.WriteLine($"Date:{post.PublishDateTime}");
+        }
+
+        private void ViewBlogPosts()
+        {
+            //List<Post> posts = _postRepository.GetByAuthor(_authorId);
```
Comment (joshdbarton) [inline TabloidCLI/UserInterfaceManagers/PostDetailManager.cs:69]: yeah, here again, this code is from the other repo, if you need it later, you can get it from there.

**Review by joshdbarton — CHANGES_REQUESTED** [review]

File: TabloidCLI/Repositories/PostRepository.cs (hunk 3 above)
Comment (karlarn) [reply TabloidCLI/Repositories/PostRepository.cs:77]: I can remove it if you want. I left this in for when the next tickets are going to need the FKs.

**Review by joshdbarton — DISMISSED** [review]

**Review by joshdbarton — APPROVED** [review]

**Review by ToryT23 — APPROVED** [review]
great

**Review by jandershock — APPROVED** [review]

**Pull request merged**